### ISSUE TITLE
Alleviate PageRetriever limitations

### ISSUE THREAD

**Issue by JanLenoch** [Kentico/xperience-training-13#78]
### Who

- [ ] Anonymous visitor
- [ ] Patient
- [ ] Doctor
- [ ] Content editor
- [ ] Marketer
- [ ] Administrator
- [x] Developer

### Needs what

* Remove explicit culture parameters from signatures of repository methods
* Add a query service that uses DocumentHelper instead of PageRetriever
* Add some basic support for caching

### Why

PageRetriever was designed to be heavily context-dependent. There are no plans to change that.

### Issue expands upon

This issue expands upon another issue(s): #`issue number`

### Blockers

This issue is blocked by the following issue(s): #`issue number`
